jstests/aggregation/bugs/server7695_isodates.js

```diff
@@ -0,0 +1,284 @@
+
+        // in the case of $dateToString the date is on property date
+        var date = value.date || value;
+        if (date.valueOf() < 0 && _isWindows() && res.code === 16422) {
+            // some versions of windows (but not all) fail with dates before 1970
+            print("skipping test of " + date.tojson() + " because system doesn't support old dates");
+            return;
+        }
+
+        if (date.valueOf()/1000 < -2*1024*1024*1024 && res.code == 16421) {
+            // we correctly detected that we are outside of the range of a 32-bit time_t
+            print("skipping test of " + date.tojson() + " because it is outside of time_t range");
+            return;
+        }
+
+        assert.eq(res.result[0].result, expResult, pipeline);
+    }
+
+    // While development, there was a bug which caused an error with $dateToString if the order of
+    // %V and %G changed, so I added this test to prevent regression.
+    testOp('$dateToString', {date: new Date("1900-12-31T23:59:59Z"), format: "%V-%G"}, "01-1901");
+    // This was failing, but it shouldn't as it is the same as above, only rotated.
+    testOp('$dateToString', {date: new Date("1900-12-31T23:59:59Z"), format: "%G-%V"}, "1901-01");
+
+
+    // 1900 is special because it's devisible by 4 and by 100 but not 400 so it's not a leap year.
+    // 2000 is special, because it's devisible by 4, 100, 400 and so it is a leap year.
+    var years = {
+        common: [
+            1900, // Starting and ending on Monday (special).
+            2002, // Starting and ending on Tuesday.
+            2014, // Starting and ending on Wednesday.
+            2015, // Starting and ending on Thursday.
+            2010, // Starting and ending on Friday.
+            2011, // Starting and ending on Saturday.
+            2006, // Starting and ending on Sunday.
+        ],
+        leap: [
+            1996, // Starting on Monday, ending on Tuesday.
+            2008, // Starting on Tuesday, ending on Wednesday.
+            1992, // Starting on Wednesday, ending on Thursday.
+            2004, // Starting on Thursday, ending on Friday.
+            2016, // Starting on Friday, ending on Saturday.
+            2000, // Starting on Saturday, ending on Sunday (special).
+            2012, // Starting on Sunday, ending on Monday.
+        ],
+        commonAfterLeap: [
+            2001, // Starting and ending on Monday.
+            2013, // Starting and ending on Tuesday.
+            1997, // Starting and ending on Wednesday.
+            2009, // Starting and ending on Thursday.
+            1993, // Starting and ending on Friday.
+            2005, // Starting and ending on Saturday.
+            2017, // Starting and ending on Sunday.
+        ],
+    };
+
+    var MONDAY = 1;
+    var TUESDAY = 2;
+    var WEDNESDAY = 3;
+    var THURSDAY = 4;
+    var FRIDAY = 5;
+    var SATURDAY = 6;
+    var SUNDAY = 7;
+
+    function padded(val) {
```

Comment:
The comment will be actually longer than the implementation itself ;-)

For the other helper methods, I already thought I had very descriptive names for the helper methods, could you please point me two helper which aren't obvious enough?